vendoring/Rust.md

```diff
@@ -0,0 +1,114 @@
+# Rust code in main
+
+Due to the current state of the Rust ecosystem, the MIR rules state that packages in main that contain Rust code should vendor their Rust dependencies rather than rely on the individual package versions, see [cpaelzer/ubuntu-mir#3](https://github.com/cpaelzer/ubuntu-mir/pull/3) for some background on the issue.
```

Comment:
It'd be nice to update this link when appropriate, how do we remember to do that?

Reply:
The link points to some historical reference, why do you think we should update this? IMO, we should rather re-write our Rust MIR rules (for doing less vendoring) and drop this reference eventually.

We can use this PR for discussing such improvements, but I have the impression we're not yet at a point where we can enforce non-vendored Rust packages. Somebody (Foundations/Toolchain?) still need to lay the groundwork of having a set of core packages available, i.e. as suggested in https://github.com/canonical/ubuntu-mir/issues/35

Reply:
Actually, I wonder if we should still enforce the vendoring mandate, today. Maybe we should still ALLOW for vendoring Rust crates, but allow package maintainers to use the `rust-*-dev` packages from the archive, too.

This would mean we need to also adopt the corresponding section in the README: "The rust ecosystem currently isn't yet considered stable enough for classic lib dependencies and transitions in main; therefore the expectation for those packages is to vendor (and own/test) all dependencies (except those provided by the rust runtime itself)."

Reply:
@setharnold In order to unblock this PR, I've added a TODO item to the PR description of https://github.com/canonical/ubuntu-mir/issues/35 So we can remember to update the section of this document, once we have proper Rust dependencies available in the archive.